include/openmc/shared_array.h

```diff
@@ -107,6 +107,9 @@ class SharedArray {
   //! \param size The new size of the container
   void resize(int64_t size) { size_ = size; }
 
+  //! Return whether the array is full
+  bool full() { return size_ == capacity_; }
```

Comment:
Could maybe mark as const, but not particularly necessary since we're not really consistent with marking methods as const elsewhere.

Reply:
Good point -- I'll add that in